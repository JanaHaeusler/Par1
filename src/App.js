import styled from 'styled-components/macro'
import useFormData from './hooks/useFormData'
import SaveGameForm from './components/SaveGameForm'
import GameCardsList from './components/GameCardsList'


function App() {

  const { gameProfile, setGameProfile, savedGameProfiles, setSavedGameProfiles } = useFormData()
 
  return (
    <AppWrapper>
      <SaveGameForm gameProfile={gameProfile} setGameProfile={setGameProfile} savedGameProfiles={savedGameProfiles} setSavedGameProfiles={setSavedGameProfiles} />
      <GameCardsList savedGameProfiles={savedGameProfiles} />
    </AppWrapper>
  );
}

export default App;

const AppWrapper = styled.div`
  display: grid;
  grid-gap: 20px;
`
