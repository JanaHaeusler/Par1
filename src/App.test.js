import { render, screen } from '@testing-library/react';
import App from './App';


describe('App', () => {
  it('renders components', () => {
      const { container } = render(<App/>)
      expect(container.firstChild).toMatchSnapshot()
  })
})


// test('renders learn react link', () => {
//   render(<App />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });
