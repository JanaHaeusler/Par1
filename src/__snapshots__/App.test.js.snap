// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders components 1`] = `
.c3 {
  padding: 7px;
  border-style: none;
  border-radius: 5px;
  box-shadow: 3px 3px 10px var(--primary-dark);
  background-color: var(--secondary-dark);
  color: var(--text-light);
  font-size: 1rem;
}

.c1 {
  padding: 0 10px 10px 10px;
  display: grid;
  place-items: center;
  border-radius: 10px;
  box-shadow: 0 0 10px var(--primary-medium);
  background-color: var(--primary-medium);
  color: var(--text-light);
  font-size: 1rem;
}

.c1 Button {
  margin-top: 15px;
  margin-bottom: 5px;
}

.c2 {
  margin: 0;
  padding: 0;
  border: none;
  display: grid;
  width: 100%;
}

.c2 label {
  margin-top: 10px;
  margin-bottom: 5px;
  font-family: 'Raleway',sans-serif;
}

.c2 input {
  padding: 5px;
  border-style: none;
  border-radius: 5px;
  color: var(--primary-dark);
  font-family: 'Montserrat',sans-serif;
}

.c4 {
  display: grid;
  gap: 10px;
}

.c0 {
  display: grid;
  grid-gap: 20px;
}

<div
  class="c0"
>
  <form
    class="c1"
  >
    <fieldset
      class="c2"
    >
      <label
        for="location"
      >
        Location
      </label>
      <input
        id="location"
        name="location"
        placeholder="Type location ..."
        type="text"
        value=""
      />
      <label
        for="date"
      >
        Date
      </label>
      <input
        id="date"
        name="date"
        type="date"
        value=""
      />
      <label
        for="players"
      >
        Player(s)
      </label>
      <input
        id="players"
        name="players"
        placeholder="John, Jane"
        type="text"
        value=""
      />
      <label
        for="winner"
      >
        Winner(s)
      </label>
      <input
        id="winner"
        name="winner"
        placeholder="Jane"
        type="text"
        value=""
      />
      <label
        for="shots"
      >
        Total Shots Winner(s)
      </label>
      <input
        id="shots"
        name="shots"
        placeholder="38"
        type="text"
        value=""
      />
    </fieldset>
    <button
      class="c3"
    >
      âœ“ Save
    </button>
  </form>
  <div
    class="c4"
  />
</div>
`;
